escrow/license_contract.proto

```diff
@@ -0,0 +1,184 @@
+    //    //to check for authorizations on Signature.
+    rpc DecrementUsage (UpdateLicenseUsageRequest) returns (UsageResponse) {
+    }
+
+
+}
+
+
+message CallerAuthentication {
+    //Signature will compromise the following data:
+    // (_method name, current block, user address) signed by private key
+    // Example ("_GetLicense", "block_number") OR ("_CreateLicense", "block_number")
+    bytes signature = 1;
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+}
+
+message UpdateLicenseUsageRequest {
+    CallerAuthentication auth = 1;
+    uint64 channel_id = 2;
+    string service_id = 3;
+
+}
+
+message LicenseCreateRequest {
+    CallerAuthentication auth = 1;
+    //the channel exclusively created for this license needs to be passed
+    uint64 channel_id = 2;
+    //you can pass the list of Authorized addresses to also be allowed to use the License.
+    repeated string user_addresses = 4;
+}
+
+message LicenseFilterByRequest {
+    CallerAuthentication auth = 1;
+    enum Operation {
+        service = 0;
+        Renew = 1;
+    }
+
+
+}
+message LicenseReadRequest {
+    CallerAuthentication auth = 1;
+    //the channel exclusively created for this license needs to be passed
+    uint64 channel_id = 2;
+}
+
+//Based on the given  service Ids passed all licenses associated with the user who has signed will be sent back
+message LicenseProviderReadRequest {
+    CallerAuthentication auth = 1;
+    repeated string service_id = 2;
+
+}
+
+message LicenseUpdateRequest {
+    //Caller authentication data
+    CallerAuthentication auth = 1;
+
+    uint64 channel_id = 2;
+
+    enum Operation {
+        Cancel = 0;
+        Renew = 1;
+    }
+    Operation operation = 3;
+
+    string service_id = 4;
+
+    string license_id = 5;
+}
+
+message DiscountData {
+    uint64 percentage_discount = 1;
+    string license_type = 2;
+}
+
+message UsageResponse {
+    enum UsageType {
+        TRACK_CALLS_MADE = 0;
+        TRACK_AMOUNT_USED = 1;
+    }
+    uint64 planned = 1;
+    uint64 used = 2;
+    uint64 refund = 3;
+    UsageType usage = 4;
+
+}
+
+message ChannelResponse {
+    uint64 channel_id = 1;
+
+    uint64 channel_nonce = 2;
+
+    bytes last_signed_amount = 3;
+
+    //indicative of the Channel Expiry in block number
+    uint64 channel_expiry = 5;
+}
+
+
+message PricingPeriodResponse {
+    uint64 period_in_days = 1;
+    string license_id = 2;
+    uint64 max_calls = 3;
+    uint64 fee_in_cogs = 4;
+    string plan_name = 5;
+    string details_url = 6;
+
+}
+
+
+message SubscriptionTypeResponse {
+    repeated PricingPeriodResponse pricing_period = 1;
+}
+
+
+message LicenseTypesSupportedResponse {
+    repeated SubscriptionTypeResponse subscriptions = 1;
+}
+
+message LicenseDataResponse {
+    ChannelResponse channel = 1;
+    string license_type = 2;
```

Comment:
What is `license_type` and how does it differs from `license_id`?

Reply:
license type is to tell you if this is a subscription / Tier pricing 
Id is more interal to track all the types available and point to the one of interest while creating a license